ci/lint/check_cpp_files_inclusion.py

```diff
@@ -38,18 +14,19 @@ def check_ray_core_inclusion(fname: str):
         ImportError: if incorrect inclusion is detected.
     """
     # Exclude protobuf, which requires absolution path for compilation.
-    cmd = f"grep 'src/ray' {fname} | grep -v 'pb'"
-    grep_res = subprocess.check_output(cmd, shell=True, text=True)
-    if grep_res:
-        raise ImportError(f"{fname} has invalid header file inclusion: {grep_res}")
+    pattern = re.compile(r"^(?=.*src/ray)(?!.*pb)")
+
+    with open(fname, "r") as file:
+        for line in file:
+            if pattern.search(line):
+                raise ImportError(f"{fname} has invalid header file inclusion: {line}")
```

Comment:
Use `file.read()` to read the whole file, and then use multiline mode for `re`. See https://docs.python.org/3/library/re.html#re.M for details.

Reply:
Updated.